### PR TITLE
Takeover of PR #2718 by @davidpendergast

### PR DESCRIPTION
**Pull request by MyreMylar**
Original PR here: #2718 

- Added some detail back in about dirty rect technique and situations it might be useful.
- Added some general optimisation advice as well.

### CODE REVIEW

File: docs/reST/tut/newbieguide.rst [hunk 1]
```diff
@@ -327,7 +329,7 @@ still be sitting in the event queue, however, waiting to be retrieved and
 processed.
 
 The lesson is: choose the system that meets your requirements.  If you don't
-have much going on in your loop -- say you're just sitting in a ``while True``
+have much going on in your loop -- say you're just sitting in a ``while 1``
```
Comment (ankith26) [inline docs/reST/tut/newbieguide.rst:332]: Why is this change being made? IMHO `while True` seems more pythonic

Comment (ankith26) [inline docs/reST/tut/newbieguide.rst:332]: Why is this change being made? IMHO `while True` seems more pythonic

**Review by ankith26 — APPROVED** [review]
Overall, good PR! 👍 Left a minor review

File: docs/reST/tut/newbieguide.rst (hunk 1 above)
Comment (MyreMylar) [reply docs/reST/tut/newbieguide.rst:332]: I agree, most of this is just copied from the original PR.

File: docs/reST/tut/newbieguide.rst [hunk 2]
```diff
@@ -19,11 +19,11 @@ or **How I learned to stop worrying and love the blit.**
 Pygame_ is a python wrapper for SDL_, written by Pete Shinners.  What this
 means is that, using pygame, you can write games or other multimedia
 applications in Python that will run unaltered on any of SDL's supported
-platforms (Windows, Unix, Mac, BeOS and others).
+platforms (Windows, Linux, Mac, and others).
 
 Pygame may be easy to learn, but the world of graphics programming can be
 pretty confusing to the newcomer.  I wrote this to try to distill the practical
-knowledge I've gained over the past year or so of working with pygame, and its
+knowledge I've gained over the past year or so of working with pygame, and it's
```
Comment (Starbuck5) [inline docs/reST/tut/newbieguide.rst:26]: Should be `its` still.

File: docs/reST/tut/newbieguide.rst [hunk 3]
```diff
@@ -123,93 +127,89 @@ output file had the same pixel format as the input file.  If you're writing a
...
+Instead of updating the whole screen every frame, only the parts that changed since
+the last frame are updated.  You do this by keeping track of those rectangles in a list,
+then calling ``update(the_dirty_rectangles)`` at the end of the frame.  In detail
+for a moving sprite:
 
  * Blit a piece of the background over the sprite's current location, erasing it.
  * Append the sprite's current location rectangle to a list called dirty_rects.
  * Move the sprite.
- * Draw the sprite at its new location.
+ * Draw the sprite at it's new location.
```
Comment (Starbuck5) [inline docs/reST/tut/newbieguide.rst:170]: should be `its` still

File: docs/reST/tut/newbieguide.rst [hunk 4]
```diff
@@ -123,93 +127,89 @@ output file had the same pixel format as the input file.  If you're writing a
...
 
- * It only works on some platforms. Windows machines can usually get hardware surfaces if you ask for them. Most other platforms can't. Linux, for example, may be able to provide a hardware surface if X4 is installed, if DGA2 is working properly, and if the moons are aligned correctly. If a hardware surface is unavailable, SDL will silently give you a software surface instead.
+**HWSURFACE and DOUBLEBUF**
 
- * It only works fullscreen.
+These :func:`.display.set_mode()` flags do nothing in pygame versions 2.0.0 and
+later (you can check the docs if you don't believe me)!  There's no reason to
+use them anymore.  Even in pygame 1, their effects were pretty nuanced and
+generally misunderstood by most pygame users.  They were never magic speed-up
+flags, unfortunately.
```
Comment (Starbuck5) [inline docs/reST/tut/newbieguide.rst:198]: Sadly, I've been wrong about this. Doublebuf still has a use for OpenGL stuff. I've gotta open a PR to correct the docs where I said doublebuf is obsolete.

However, this is right that it is not a magic speed up flag!

**Review by Starbuck5 — APPROVED** [review]
Thanks for taking this over, thanks to Ghast for the original PR.

Left a couple small reviews that should be handled before merging.

File: docs/reST/tut/newbieguide.rst (hunk 4 above)
Comment (MyreMylar) [reply docs/reST/tut/newbieguide.rst]: :) I adjusted this section to reflect truth and this sentiment.

**Pull request merged**